be/src/exec/pipeline/query_context.h

```diff
@@ -143,8 +143,8 @@ class QueryContext : public std::enable_shared_from_this<QueryContext> {
     void release_workgroup_token_once();
 
     // Some statistic about the query, including cpu, scan_rows, scan_bytes
-    int64_t mem_cost_bytes() const { return _mem_tracker->peak_consumption(); }
-    int64_t current_mem_usage_bytes() const { return _mem_tracker->consumption(); }
+    int64_t mem_cost_bytes() const { return _mem_tracker == nullptr ? 0 : _mem_tracker->peak_consumption(); }
+    int64_t current_mem_usage_bytes() const { return _mem_tracker == nullptr ? 0 : _mem_tracker->consumption(); }
```

Comment:
My question is why this can be accessed before _mem_tracker is created.
Why not initialize mem_tracker when this object is created?

Reply:
Because the prepare will go through many steps, each step only initialize some of the fields. If the `show proc '/current_queries'` are executed in the middle of these step, it may get a queryContext with some fileds not well  initialized.

Reply:
Why not create the memtracker at the first step of initialization?

Reply:
Following is the main steps of prepare stage. Memtracker is initialized at the `_prepare_runtime_state` step, it replies on the fragment_context, resource_group, so it can't be initialized at the first step.

```
    {
        SCOPED_RAW_TIMER(&profiler.prepare_query_ctx_time);
        RETURN_IF_ERROR(_prepare_query_ctx(exec_env, request));
    }
    {
        SCOPED_RAW_TIMER(&profiler.prepare_fragment_ctx_time);
        RETURN_IF_ERROR(_prepare_fragment_ctx(request));
    }
    {
        SCOPED_RAW_TIMER(&profiler.prepare_runtime_state_time);
        RETURN_IF_ERROR(_prepare_workgroup(request));
        RETURN_IF_ERROR(_prepare_runtime_state(exec_env, request));
        RETURN_IF_ERROR(_prepare_exec_plan(exec_env, request));
        RETURN_IF_ERROR(_prepare_global_dict(request));
    }
    {
        SCOPED_RAW_TIMER(&profiler.prepare_pipeline_driver_time);
        RETURN_IF_ERROR(_prepare_pipeline_driver(exec_env, request));
        RETURN_IF_ERROR(_prepare_stream_load_pipe(exec_env, request));
    }
```

Reply:
I have changed the way to fix this bug, please review again.